include/swift/AST/DiagnosticConsumer.h

```diff
@@ -100,7 +100,17 @@ class DiagnosticConsumer {
                                 const DiagnosticInfo &Info) = 0;
 
   /// \returns true if an error occurred while finishing-up.
-  virtual bool finishProcessing(SourceManager &) { return false; }
+  virtual bool finishProcessing() { return false; }
+
+  /// In batch mode, any error causes failure for all primary files, but
+  /// Xcode will only see an error for a particular primary in that primary's
```

Comment:
While in this case the motivation involves Xcode, the explanation of who-sees-what here applies to anyone consuming .dia files.

Reply:
I really don't like how something intrinsic to FileSpecificDiagnosticConsumer is leaking out into the other consumers. (If it wasn't, you could just put this logic in finishProcessing, but we know we can't do that because that would affect single-file and WMO.)

Reply:
Fixed (not pushed yet). Thanks!